fighters/donkey/src/status/special_lw.rs

```diff
@@ -2,18 +2,42 @@ use super::*;
 
 #[status_script(agent = "donkey", status = FIGHTER_STATUS_KIND_SPECIAL_LW, condition = LUA_SCRIPT_STATUS_FUNC_STATUS_MAIN)]
 unsafe fn special_lw_main(fighter: &mut L2CFighterCommon) -> L2CValue {
-    let motion;
-    let kinetic;
-    if fighter.global_table[SITUATION_KIND].get_i32() != *SITUATION_KIND_GROUND {
-        motion = Hash40::new("special_air_lw");
-        kinetic = *FIGHTER_KINETIC_TYPE_FALL;
-        WorkModule::enable_transition_term(fighter.module_accessor, *FIGHTER_STATUS_TRANSITION_TERM_ID_LANDING);
-    }
-    else {
-        motion = Hash40::new("special_lw_start");
-        kinetic = *FIGHTER_KINETIC_TYPE_GROUND_STOP;
-        WorkModule::unable_transition_term_group(fighter.module_accessor, *FIGHTER_STATUS_TRANSITION_GROUP_CHK_AIR_LANDING);
+
+    // if you are grounded, pick up heavy item/spawn barrel
+    if fighter.global_table[SITUATION_KIND].get_i32() == *SITUATION_KIND_GROUND {
+
+        // if you aren't already holding an item, try to pick up one nearby
+        if !ItemModule::is_have_item(fighter.module_accessor, 0) {
+            ItemModule::pickup_item(fighter.module_accessor, ItemSize{_address: *ITEM_SIZE_HEAVY as u8}, *FIGHTER_HAVE_ITEM_WORK_MAIN, *ITEM_TRAIT_ALL, QuickItemTreatType{_address: *QUICK_ITEM_TREAT_TYPE_FORCE_HAVE as u8}, ItemPickupSearchMode{_address: *ITEM_PICKUP_SEARCH_MODE_NORMAL as u8});
+        }
+        if !ItemModule::is_have_item(fighter.module_accessor, 0) {
+            ItemModule::pickup_item(fighter.module_accessor, ItemSize{_address: *ITEM_SIZE_LIGHT as u8}, *FIGHTER_HAVE_ITEM_WORK_MAIN, *ITEM_TRAIT_ALL, QuickItemTreatType{_address: *QUICK_ITEM_TREAT_TYPE_FORCE_HAVE as u8}, ItemPickupSearchMode{_address: *ITEM_PICKUP_SEARCH_MODE_NORMAL as u8});
+        }
+        
+        // if you still arent holding an item, try to spawn a barrel
+        if !ItemModule::is_have_item(fighter.module_accessor, 0) {
+            VarModule::on_flag(fighter.object(), vars::donkey::instance::DID_SPAWN_BARREL);
+            let itemmanager = smash2::app::ItemManager::instance().unwrap();
+            let barrel_count = smash2::app::ItemManager::get_num_of_ownered_item(
```

Comment:
This may potentially cause DK to not be able to spawn a barrel if first picking up and throwing another DK's barrel, but I haven't tested that

Reply:
Probably correct, as you then assume ownership of that barrel.